relayer-engine/src/executor/executorHarness.ts

```diff
@@ -308,22 +316,22 @@ async function spawnWalletWorker(
         continue;
       }
       const actionWithCont = actionQueue.dequeue();
-      logger.info(`Relaying action for plugin ${actionWithCont.pluginName}...`);
+      logger.info(`Relaying action for plugin ${actionWithCont.pluginName}...`, {action: actionWithCont});
 
       try {
         const result = await actionWithCont.action.f(
           walletToolBox,
           workerInfo.targetChainId,
         );
-        logger.info(`Action ${actionWithCont.pluginName} completed`);
+        logger.info(`Action ${actionWithCont.pluginName} completed`, {action: actionWithCont});
         actionWithCont.resolve(result);
       } catch (e) {
-        logger.error(e);
-        logger.warn(`Unexpected error while executing chain action:`);
+        logger.warn(`Unexpected error while executing chain action:`, {action: actionWithCont});
+        logger.error("", { action: actionWithCont, error: e });
```

Comment:
Why log `actionWithCont` twice? It just adds additional logs with no real value. Why not something like this instead?

```suggestion
        logger.error("Unexpected error while executing chain action", { action: actionWithCont, error: e });
```

Redundancy is redundant :)